src/options.c

```diff
@@ -321,24 +321,54 @@ static struct opt_entry opt_list[] = {
 
 #if HAVE_LIBGMP || HAVE_INT128 || HAVE___INT128 || HAVE___INT128_T
 #define PRINCE_USAGE \
-"--prince[=FILE]            PRINCE mode, read words from FILE\n"
+"--prince[=FILE]            PRINCE mode, read words from FILE\n" \
+"--prince-loopback[=FILE]   fetch words from a .pot file\n" \
+"--prince-elem-cnt-min=N    minimum number of elements per chain (1)\n" \
+"--prince-elem-cnt-max=[N]  maximum number of elements per chain (negative N is\n" \
+"                           relative to word length) (8)\n" \
+"--prince-skip=N            initial skip\n" \
+"--prince-limit=N           limit number of candidates generated\n" \
+"--prince-wl-dist-len       calculate length distribution from wordlist\n" \
+"--prince-wl-max=N          load only N words from input wordlist\n" \
+"--prince-case-permute      permute case of first letter\n" \
+"--prince-mmap              memory-map infile (not available with case permute)\n" \
+"--prince-keyspace          just show total keyspace that would be produced\n" \
+"                           (disregarding skip and limit)\n"
 #else
 #define PRINCE_USAGE ""
 #endif
 
-#define JOHN_USAGE	  \
+#ifdef HAVE_FUZZ
+#define FUZZ_USAGE \
+"--fuzz[=DICTFILE]         fuzz formats' prepare(), valid() and split()\n" \
+"--fuzz-dump[=FROM,TO]     dump the fuzzed hashes between FROM and TO to file pwfile.format\n"
+#else
+#define FUZZ_USAGE ""
+#endif
+
+#define JOHN_BANNER	  \
 "John the Ripper " JTR_GIT_VERSION _MP_VERSION DEBUG_STRING ASAN_STRING UBSAN_STRING " [" JOHN_BLD "]\n" \
 "Copyright (c) 1996-2019 by " JOHN_COPYRIGHT "\n" \
 "Homepage: https://www.openwall.com/john/\n" \
 "\n" \
-"Usage: %s [OPTIONS] [PASSWORD-FILES]\n" \
+"Usage: %s [OPTIONS] [PASSWORD-FILES]\n"                                                  \
+
+#define JOHN_USAGE    \
+"--help                     print usage summary\n" \
+"--config=FILE              use FILE instead of john.conf or john.ini\n" \
 "--single[=SECTION[,..]]    \"single crack\" mode, using default or named rules\n" \
 "--single=:rule[,..]        same, using \"immediate\" rule(s)\n" \
+"--single-seed=WORD[,WORD]  add static seed word(s) for all salts in single mode\n" \
+"--single-wordlist=FILE     *short* wordlist with static seed words/morphemes\n" \
+"--single-user-seed=FILE    wordlist with seeds per username (user:password[s] format)\n" \
+"--single-retest-guess=B    override config for SingleRetestGuess (bool: Y/N)\n" \
 "--wordlist[=FILE] --stdin  wordlist mode, read words from FILE or stdin\n" \
 "                  --pipe   like --stdin, but bulk reads, and allows rules\n" \
+"--mem-file-size=SIZE       size threshold for wordlist preload (default %u MB)\n" \
 "--loopback[=FILE]          like --wordlist, but extract words from a .pot file\n" \
-"--dupe-suppression         suppress all dupes in wordlist (and force preload)\n" \
-PRINCE_USAGE \
+"--dupe-suppression         suppress all dupes in wordlist (and force preload)\n"                        \
```

Comment:
...and ilkewise these should be `--wordlist-dupe-suppression` and `--wordlist-loopback`.

PRINCE option is great with tab completion in that you can quickly find all options by saying `--prince=` and hit tab. Renaming the mentioned options would give the same advantage.

Reply:
I covered these in #4399 as well.